freqtrade/exchange/__init__.py

```diff
@@ -546,13 +546,23 @@ def refresh_latest_ohlcv(self, pair_list: List[Tuple[str, str]]) -> List[Tuple[s
 
         input_coroutines = []
 
+        logger.info(f"@@@ {self._exchange_internals_options}")
+
         # Gather coroutines to run
         for pair, ticker_interval in set(pair_list):
+            if not ((pair, ticker_interval) in self._klines):
+                logger.info(f"Ohlcv data for ({pair}, {ticker_interval}) not in klines.")
             if (not ((pair, ticker_interval) in self._klines)
                     or self._now_is_time_to_refresh(pair, ticker_interval)):
-                input_coroutines.append(self._async_get_candle_history(pair, ticker_interval))
+                logger.info(f"Refreshing ohlcv data for ({pair}, {ticker_interval}) ...")
+                _LIMIT = 100
+                limit = self._exchange_internals_options.get('max_api_request_count', _LIMIT)
```

Comment:
I am not sure of defaulting to 100 here.
The default is "None" - which returns the "exchange default" - which can be 500 for binance, 700 for kraken, or 5000 for Bittrex (??) - testing a 5m timeframe of ETH/BTC.

Defaulting to 100 will be a heavy change to user experience, and break a ton of peoples strategies using a sma/ema of 200 ...

Reply:
Well, guys (sorry, very brief, a bit busy now):

1) Of course it a Work in Progress. The whole BitMEX support will be WIP for a long time, there are lot's of issues to be resolved. I go step by step. By the meantime FT cannot even fetch candle data to create dataframe for the strategy. Then I'm going to go though trades through sales, then historical data, Edge, stoplosses, fees, etc., there are many things to do. Sure it's all WIP and will be for near future.

2) As I see, I need to modify the bahaviour of the Exchange methods. On a exchange-specific manner. I decided to make a subclass for BitMEX to have place for exchange specific options ('features') and (in the future) for exchange-specific methods.

3) @xmatthias If you take a look, I implemented approach similar to your 'ft_has'. I only called the subclass differentely, 'exchange_internals_options'. Because my idea was that these options could be merged (in the future) with config, with 'internals' subdict in the 'exchange' dict. That's the only difference between our ideas.

4) What I committed here after first commit -- it's just a dirty temporary wip implementation from a local branch. I wanted just a new subclass, but @xmatthias said he did not see a reason for this, so further commits are just for illustration purposes...

5) _LIMIT=100 there was just a temporary WIP value, a part of WIP support for pagination. I guess there should be 2 values, something like EXCHANGE_DEFAULT_RESPONSE_LEN = 100, EXCHANGE_MAX_RESPONSE_LEN = 500 (in constants, used in the absence of exchange specific options in the exchange subclass and/or config) and exchange_default_response_len, exchange_max_response_len options specific for Exchange, defined somewhere in the exchange subclass and/or config.

I think we should better close it for now, I'll wait for new ft_has conception committed into develop and reimplement it on more accurate and clean way using ft_has...

Reply:
One note: if you expect the PR to be wip for some time, add a [WIP] to the title so we know about it.

Ignore my review then - I assumed it's closer to being merged than it really is, and i tend to leave "notes" i see on open Pr's from time to time ... 

to be clear, my "ft_has" is not set in stone, maybe it's going to be changed, it's close to "ccxt.has" - but noone says that's the correct way.